### PR TITLE
Remove duplicate metric for absent cache

### PR DESCRIPTION
**Pull request by dbw9580**
Remove duplicate metric `Master.UfsAbsentPathCacheSize`. 

Rename `UfsAbsentPathCacheQueueSize` to `AbsentPathCacheQueueSize` to be consistent with already existing metric name.

Also make other similar metrics cached.

### CODE REVIEW

**Comment by dbw9580** [comment]
This depends on #15356. will rebase once that PR is merged.

**Review by jiacheliu3 — APPROVED** [review]
LGTM, good to merge once the rebase is done. Thx!

**Comment by jiacheliu3** [comment]
oops @dbw9580 
```
Error: 5.667 [ERROR] Failures: 
Error: 5.667 [ERROR] alluxio.master.file.meta.AsyncUfsAbsentPathCacheTest.metricCacheHitsAndMisses
Error: 5.668 [ERROR]   Run 1: AsyncUfsAbsentPathCacheTest.metricCacheHitsAndMisses:282 expected:<2> but was:<1>
Error: 5.668 [ERROR]   Run 2: AsyncUfsAbsentPathCacheTest.metricCacheHitsAndMisses:282 expected:<2> but was:<1>
04:10:35.668 [INFO] 
Error: 5.668 [ERROR] alluxio.master.file.meta.AsyncUfsAbsentPathCacheTest.metricCacheSize
Error: 5.668 [ERROR]   Run 1: AsyncUfsAbsentPathCacheTest.metricCacheSize:261 expected:<2> but was:<1>
Error: 5.668 [ERROR]   Run 2: AsyncUfsAbsentPathCacheTest.metricCacheSize:261 expected:<2> but was:<1>
```

**Comment by dbw9580** [comment]
> oops @dbw9580 
> ```
> Error: 5.667 [ERROR] Failures: 
> Error: 5.667 [ERROR] alluxio.master.file.meta.AsyncUfsAbsentPathCacheTest.metricCacheHitsAndMisses
> Error: 5.668 [ERROR]   Run 1: AsyncUfsAbsentPathCacheTest.metricCacheHitsAndMisses:282 expected:<2> but was:<1>
> Error: 5.668 [ERROR]   Run 2: AsyncUfsAbsentPathCacheTest.metricCacheHitsAndMisses:282 expected:<2> but was:<1>
> 04:10:35.668 [INFO] 
> Error: 5.668 [ERROR] alluxio.master.file.meta.AsyncUfsAbsentPathCacheTest.metricCacheSize
> Error: 5.668 [ERROR]   Run 1: AsyncUfsAbsentPathCacheTest.metricCacheSize:261 expected:<2> but was:<1>
> Error: 5.668 [ERROR]   Run 2: AsyncUfsAbsentPathCacheTest.metricCacheSize:261 expected:<2> but was:<1>
> ```

duh, the gauges are now cached. The count is stale.

**Comment by dbw9580** [comment]
The added `mockito-inline` breaks existing tests.

https://javadoc.io/static/org.mockito/mockito-core/4.1.0/org/mockito/Mockito.html#Mocking_Final

```
Error: 5.229 [ERROR] alluxio.master.block.JvmSpaceReviewerTest.reviewJvmUsage  Time elapsed: 1.72 s  <<< ERROR!
org.mockito.exceptions.misusing.MissingMethodInvocationException: 

when() requires an argument which has to be 'a method call on a mock'.
For example:
    when(mock.getArticles()).thenReturn(articles);

Also, this error might show up because:
1. you stub either of: final/private/equals()/hashCode() methods.
   Those methods *cannot* be stubbed/verified.
   Mocking methods declared on non-public parent classes is not supported.
2. inside when() you don't call method on mock but on some other object.

	at alluxio.master.block.JvmSpaceReviewerTest.reviewJvmUsage(JvmSpaceReviewerTest.java:50)
```

**Comment by dbw9580** [comment]
@jiacheliu3 fixed tests, PTAL.

File: core/server/master/src/test/java/alluxio/master/file/meta/AsyncUfsAbsentPathCacheTest.java
Comment (jiacheliu3) [inline core/server/master/src/test/java/alluxio/master/file/meta/AsyncUfsAbsentPathCacheTest.java:266]: I'm thinking this mock may be fragile because it relies on how the gauge looks at time. Since you have the static variable already, can you set this to a super small number in the test and sleep between peeking at the cached gauge, to make sure the cache goes away?

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Added a comment on the UT, PTAL :)

File: core/server/master/src/test/java/alluxio/master/file/meta/AsyncUfsAbsentPathCacheTest.java
Comment (dbw9580) [reply core/server/master/src/test/java/alluxio/master/file/meta/AsyncUfsAbsentPathCacheTest.java:266]: You mean make `AsyncUfsAbsentPathCache.METRIC_CACHE_TIMEOUT_SEC` mutable? And set it in UTs?

Comment (jiacheliu3) [reply core/server/master/src/test/java/alluxio/master/file/meta/AsyncUfsAbsentPathCacheTest.java:266]: oops it's impossible for Powermock or reflection to update a static final constant at runtime. So you can do this:

```
int getCachedGaugeTimeoutSec() {
  return 2;
}
```
then
```
class TestAsyncUfsAbsentPathCache extends AsyncUfsAbsentPathCache {
  @Override
  int getCachedGaugeTimeoutSec() {
    // not sure what happens if you return 0 here
    // but if caching 0 sec is invalid argument, you can make the method based on millisecond and return a 1ms
  }
}
```

Comment (dbw9580) [reply core/server/master/src/test/java/alluxio/master/file/meta/AsyncUfsAbsentPathCacheTest.java:266]: Sounds good.

File: core/server/master/src/main/java/alluxio/master/file/meta/AsyncUfsAbsentPathCache.java
Comment (dbw9580) [inline core/server/master/src/main/java/alluxio/master/file/meta/AsyncUfsAbsentPathCache.java:52]: Have to remove final-ness to make the class inheritable.

File: core/server/master/src/test/java/alluxio/master/file/meta/AsyncUfsAbsentPathCacheTest.java
Comment (jiacheliu3) [inline core/server/master/src/test/java/alluxio/master/file/meta/AsyncUfsAbsentPathCacheTest.java:338]: nit: add a comment why this class is needed

**Review by jiacheliu3 — APPROVED** [review]
LGTM with a minor comment, thx!

File: core/server/master/src/test/java/alluxio/master/file/meta/AsyncUfsAbsentPathCacheTest.java
Comment (dbw9580) [reply core/server/master/src/test/java/alluxio/master/file/meta/AsyncUfsAbsentPathCacheTest.java:341]: added in 27abf0c3

**Comment by dbw9580** [comment]
@jiacheliu3 ready to merge?

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks!

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Comment by jiacheliu3** [comment]
@Xenorith this is a small metrics change unrelated to performance. Could you include this in 2.8 if possible? Thx!

**Comment by Xenorith** [comment]
i will not cherry pick this for 2.8 unless there's another testing cycle

**Comment by yuzhu** [comment]
alluxio-bot, cherry-pick this to branch-2.8 please

**Pull request merged**
